### ISSUE TITLE
Google Pixel stops instantly when trying to capture video

### ISSUE THREAD

**Issue by Boyan1912** [jamesmontemagno/MediaPlugin#514]
## Bug Information
Version Number of Plugin: 3.1.3
Device Tested On: Google Pixel
Device Android Version: 8.1.0
Simulator Tested On: does not reproduce on simulators (Genymotion, Android Studio simulators and others)
Version of VS: Visual Studio Enterprise 2015, Version 14.0.25431.01 Update 3
Version of Xamarin: 4.9.0.752
Versions of Xamarin.Android SDK: 8.2.0.16

### Expected Behavior
Shooting video should be uninterrupted until either stopped by the user or reached the desired size.

### Actual Behavior
When I tap on the button to start recording the device makes the sound for the beginning of the record and right after that the sound for the end of the recording and the recording stops. It generates a MediaFile with the length of 0 seconds. When you play it it looks like a single picture.
Also tried without the DesiredSize property of StoreVideoOptions.

### Code snippet
var mediaOptions = new StoreVideoOptions
                            {
                                Directory = "HCS",
                                Name = $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}.mp4",
                                SaveToAlbum = true,
                                Quality = VideoQuality.Medium,
                                DesiredSize = 45 * 1000000
                                CompressionQuality = 0,
                            };

                            MediaFile video = await CrossMedia.Current.TakeVideoAsync(mediaOptions);

**Comment by jamesmontemagno** [comment]
I tried:
```
var mediaOptions = new StoreVideoOptions
{
Directory = "HCS",
Name = $"{DateTime.UtcNow.ToString("yyyyMMddHHmmss")}.mp4",
SaveToAlbum = true,
Quality = VideoQuality.Medium,
DesiredSize = 45 * 1000000
CompressionQuality = 0,
};
```

When you do this the "DesiredLength" is set to 600, which should be 600 seconds according to documentation, however on the Pixel it is not respected. I would have to not set this specific flag perhaps. 

The issue seems to be that previously they wanted an int.. but now want a double. I have a new beta nuget to try: 3.2.0.246-beta

I don't have any other phones except for my pixel here to test, so please test this out and i will push to stable when validated.

**Comment by philmoz** [comment]
3.2.0.246-beta - records video on Pixel phone; but does not stop when 'DesiredLength' value is reached.

**Comment by jamesmontemagno** [comment]
Not all phones and camera apps actually respect the desired length. I think the new camera app is one f them.

**Comment by jamesmontemagno** [comment]
what about other phones?

**Comment by softsan** [comment]
Tested it on android 6(Moto G)  and Pixel 8.1, but it does not respect the desired length on both of the device.

**Comment by caseyniemann** [comment]
@jamesmontemagno confirming we tested multiple android devices and the desired length is not respected

**Comment by jamesmontemagno** [comment]
Did they work previously? Interesting.... Did it work previously on 8.0?

**Comment by jamesmontemagno** [comment]
What I will probably will do is turn off the functionality on 8.1+ devices.

**Comment by softsan** [comment]
@jamesmontemagno  - Also is there any possibility (or may be feature suggestion) for showing countdown when taking video?  e.x. if desired length is set to 1 minute, then it should show 60,59,58,57.....1 seconds.

**Comment by jamesmontemagno** [comment]
@softsan that is all dependent on the a camera app, we have no control. You can also request a max size in bytes already.

**Comment by Boyan1912** [comment]
@jamesmontemagno I don't think "desired length" was working on Android before. We are using "desired length" for iOS and  "desired size" for Android in our project.

**Comment by caseyniemann** [comment]
@jamesmontemagno confirming that "desired length" in MB is not working on Android...nor is "desired length" in time.

**Comment by jamesmontemagno** [comment]
Yes, but did it ever work is my question and on what devices? Did it stop working after this "fix" on those devices?

**Comment by BenReierson** [comment]
Confirmed that 3.2.0.246-beta now records successfully on Pixel 2. Previously it would stop immediately.

**Comment by philmoz** [comment]
3.2.0.246-beta - recorded video on Pixel phone; but whatever was fixed did not make it to 3.2.1, this still stops immediately on Pixel phone.

**Comment by jamesmontemagno** [comment]
I have rolled back 3.2.1

**Comment by jamesmontemagno** [comment]
try 4.x-pre

**Comment by philmoz** [comment]
Not fixed in 4.0.0.3-beta.

If DesiredLength > 0 on Google Pixel Phone (Oreo 8.0, API 26), then video recording stops immediately.

**Comment by jamesmontemagno** [comment]
you can try 4.0.0.4-beta, i now filter out all pixel phones
